web/src/protect/users/passwords.rs

```diff
@@ -0,0 +1,30 @@
+use crate::{extractors::authenticated_user::AuthenticatedUser, AppState};
+use axum::{
+    extract::{Path, Request, State},
+    http::StatusCode,
+    middleware::Next,
+    response::IntoResponse,
+};
+use domain::Id;
+use log::*;
+
+// checks:
+// - that the `user_id` matches the `authenticated_user.id`
+pub(crate) async fn update_password(
+    State(_app_state): State<AppState>,
+    AuthenticatedUser(authenticated_user): AuthenticatedUser,
+    Path(user_id): Path<Id>,
+    request: Request,
+    next: Next,
+) -> impl IntoResponse {
+    // check that we are only allowing authenticated users to update their own password
+    if authenticated_user.id == user_id {
+        next.run(request).await
+    } else {
+        error!(
+            "Unauthorized: user_id {} does not match authenticated_user_id {}",
```

Comment:
```suggestion
            "Unauthorized: user_id {} does not match authenticated_user_id {} when attempting to update password",
```

Just a suggestion that may help us observe this security-sensitive code more closely when in production. I'm a little nervous about our login handling since we wrote it ourselves.

Reply:
Updated! fd1af337616628751235f836a2f178f8dfc63d9d